src/webots/vrml/WbNode.cpp

```diff
@@ -972,13 +974,43 @@ void WbNode::writeParameters(WbVrmlWriter &writer) const {
     parameter->write(writer);
 }
 
+bool WbNode::isUrdfLinkRoot() const {
+  return findSFString("name") ? true : false;
+}
+
+WbNode *WbNode::findUrdfLinkRoot() const {
+  WbNode *parentRoot = parent();
+  while (!parentRoot->isUrdfLinkRoot())
+    parentRoot = parentRoot->parent();
```

Comment:
That looks dangerous, for example, if you call `Transform::findUrdfLinkRoot` you will end up in an infinite loop.
We should add an exit condition (and return NULL in that case).